content/docs/reference/oauth-integration.md

```diff
@@ -0,0 +1,80 @@
+
+### 1. Initiating the OAuth flow
+
+To initiate the OAuth flow, you will need to generate an authorization URL. To do that you can redirect your users to `https://oauth2.neon.tech/oauth2/auth` while passing the following query parameters:
+
+- `client_id`: your OAuth application's ID
+- `redirect_uri`: the full URL that Neon should redirect users to after authorizing your app. This URL should match at least one of the callback URLs you provided when applying to become a partner.
+- `scope`: This is a space-separated list of scopes that you want to request access to. Example value: `urn:neoncloud:projects:create urn:neoncloud:projects:read urn:neoncloud:projects:update urn:neoncloud:projects:delete`
+- `response_type`: This should be set to `code` to indicate that you are using the [Authorization Code grant type]().
+- `code_challenge`: This is a random string that will be used to verify the integrity of the authorization code.
+- `state`: This is a random string that will be returned to your callback URL. You can use this parameter to verify that the request came from your application and not from a third party.
+
+Here is an example of what the authorization URL might look like:
+
+```text
+https://oauth2.neon.tech/oauth2/auth?client_id=neon-experimental&scope=openid%20offline%20offline_access%20urn%3Aneoncloud%3Aprojects%3Acreate%20urn%3Aneoncloud%3Aprojects%3Aread%20urn%3Aneoncloud%3Aprojects%3Aupdate%20urn%3Aneoncloud%3Aprojects%3Adelete&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback%2Fneon&grant_type=authorization_code&state=H58y-rSTebc3QmNbRjNTX9dL73-IyoU2T_WNievO9as&code_challenge=99XcbwOFU6iEsvXr77Xxwsk9I0GL4c4c4Q8yPIVrF_0&code_challenge_method=S256
+```
+
+After the user is redirected to the authorization URL, they will be presented with Neon's consent screen, which will be pre-populated with the scopes you requested. The user will be able to review the scopes and authorize your application to connect their Neon account.
+
+![Neon OAuth consent screen](/docs/oauth/consent.png)
+
+### 2. Authorization code is returned to your callback URL
+
+After your user successfully completes the authorization flow, they will be redirected to the callback URL with the following query parameters appended to the URL:
+
+- `code`: an authorization code that will be exchanged for an access token
+- `scope`: the scopes that the user authorized your application to access
+- `state`: you can compare the value of this parameter with the original `state` you provided in the previous step to ensure that the request came from your application and not from a third party
+
+### 3. Exchanging the authorization code for an access token
+
+You can now exchange the authorization code returned from the previous step for an access token. To do that, you will need to send a `POST` request to `https://oauth2.neon.tech/oauth2/token` with the following parameters:
+
+- `client_id`
```

Comment:
missing description

Reply:
Should we repeat the description if it's the same as above?